hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -719,25 +735,29 @@ def pass_to_descendant(self):
         late in the process when we've got the number of intervals as far down
         as possible.
         """
-        for ex in self.each_non_trivial_example():
-            st = self.shrink_target
-            descendants = sorted(
-                set(
-                    st.buffer[d.start : d.end]
-                    for d in self.shrink_target.examples
-                    if d.start >= ex.start
-                    and d.end <= ex.end
-                    and d.length < ex.length
-                    and d.label == ex.label
-                ),
-                key=sort_key,
-            )
+        return [
+            partial(self.replace_with_descendant, i, j)
+            for i in hrange(len(self.examples))
+            for j in hrange(i + 1, len(self.examples))
+        ]
 
-            for d in descendants:
-                if self.incorporate_new_buffer(
-                    self.buffer[: ex.start] + d + self.buffer[ex.end :]
-                ):
-                    break
+    def replace_with_descendant(self, i, j):
+        assert i < j
+        if j >= len(self.examples):
+            return
+        ancestor = self.examples[i]
+        descendant = self.examples[j]
+        if (
+            descendant.start >= ancestor.end
+            or descendant.length == ancestor.length
+            or descendant.label != ancestor.label
+        ):
```

Comment:
I notice that this no longer checks that ancestor is not `.trivial`, which was previously handled by `each_non_trivial_example`.

Is this deliberate, or an oversight?

Reply:
(If the latter, there are a few other passes that previously used `each_non_trivial_example`, and could benefit from the same non-trivial check.)

Reply:
On top of that, `each_non_trivial_example` appears to now be unused, so it and its tests can be deleted if there are no plans to use it in future.